src/routes/teams.ts

```diff
@@ -0,0 +1,434 @@
+import * as express from "express";
+import { IAuthRequest } from "../config/jwt";
+import authenticate from "../middlewares/authenticate";
+import checkToken from "../middlewares/checkToken";
+import Team, { ITeamModel } from "../models/team";
+import User from "../models/user";
+
+const router = express.Router();
+
+/**
+ * GET teams with queries
+ * @param {number} contestId
+ * @param {boolean} available - only get available teams if true
+ * @param {number} begin
+ * @param {number} end
+ * @returns {Object[]} teams of given contest
+ */
+router.get(
+  "/",
+  checkToken,
+  (
+    req: {
+      query: { [key: string]: string };
+      auth: IAuthRequest;
+    },
+    res
+  ) => {
+    const query: {
+      contestId?: number;
+      available?: boolean;
+      begin?: number;
+      end?: number;
+    } = {};
+    if (req.query.contestId) {
+      query.contestId = parseInt(req.query.contestId, 10);
+    }
+    if (req.query.available === "true") {
+      query.available = true;
+    }
+
+    const begin = parseInt(req.query.begin, 10) || 0;
+    const end = parseInt(req.query.end, 10) || Number.MAX_SAFE_INTEGER;
+    const select =
+      "-_id -__v" + (req.auth.role === "root" ? "" : " -inviteCode");
+
+    Team.find(
+      { ...query, members: { $nin: req.auth.id } },
+      select,
+      (err, teams) => {
+        if (err) {
+          return res.status(500).end();
+        }
+
+        Team.find(
+          { ...query, members: { $in: req.auth.id } },
+          "-_id -__v",
+          (error, teamSelf) => {
```

Comment:
可以都统一用 `err`, 内层变量的优先级更高。
```suggestion
          (err, teamSelf) => {
```

Reply:
一般情况下，变量不 shadow 是不是更好一些，不论是之后的调用还是逻辑理解上。